### PR TITLE
Update MediaSegments directly

### PR DESCRIPTION
**Pull request by rlauuzo**
## Summary by Sourcery

Update the media segment handling by introducing a MediaSegmentUpdateManager and refactor the Entrypoint class to improve concurrency management. Convert several methods to asynchronous to enhance performance and add new configuration options for media segment updates.

New Features:
- Introduce MediaSegmentUpdateManager to handle media segment updates for episodes.

Enhancements:
- Refactor the Entrypoint class to use a SemaphoreSlim for managing analysis concurrency.
- Simplify the OnItemChanged method by consolidating logic for handling item additions and modifications.
- Convert synchronous methods to asynchronous in various tasks to improve performance and responsiveness.

Documentation:
- Add documentation for new configuration options related to media segment updates in the HTML configuration page.

### CODE REVIEW

**Comment by rlauuzo** [comment]
@sourcery-ai review

File: ConfusedPolarBear.Plugin.IntroSkipper/Manager/MediaSegmentUpdateManager.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    {
+                        _logger.LogDebug("No segments found for episode {EpisodeId}", episode.EpisodeId);
+                        continue;
+                    }
+
+                    var tasks = new List<Task>();
+                    tasks.AddRange(existingSegments.Select(s => _mediaSegmentManager.DeleteSegmentAsync(s.Id)));
+                    tasks.AddRange(newSegments.Select(s => _mediaSegmentManager.CreateSegmentAsync(s, _name)));
+
+                    await Task.WhenAll(tasks).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (jumoog) [reply ConfusedPolarBear.Plugin.IntroSkipper/Manager/MediaSegmentUpdateManager.cs:55]: 🤝

File: ConfusedPolarBear.Plugin.IntroSkipper/Configuration/configPage.html [hunk 2]
```diff
@@ -153,6 +153,29 @@ <h3 class="checkboxListLabel">Limit analysis to the following libraries</h3>
...
+                                <summary>Jellyfin Mediasegment Generation</summary>
+
+                                <br />
+                                <div class="checkboxContainer checkboxContainer-withDescription">
+                                    <label class="emby-checkbox-label">
+                                        <input id="UpdateMediaSegments" type="checkbox" is="emby-checkbox" />
+                                        <span>Update Media Segments for Newly Added Files During Scan</span>
+                                    </label>
+
+                                    <div class="fieldDescription">Enable this option to update media segments for newly added files during a scan. <strong>Warning:</strong> This should be disabled if you're using media segment providers other than Intro Skipper.</div>
```
Comment (jumoog) [inline ConfusedPolarBear.Plugin.IntroSkipper/Configuration/configPage.html:166]: I dont think the warning is needed. Nothing bad will happen if you keep it on.

Comment (rlauuzo) [reply ConfusedPolarBear.Plugin.IntroSkipper/Configuration/configPage.html:166]: It will delete all segment by other providers.

Comment (jumoog) [reply ConfusedPolarBear.Plugin.IntroSkipper/Configuration/configPage.html:166]: it maybe has an performance impact

**Review by jumoog — APPROVED** [review]

**Pull request merged**